src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the precision (1 = Short, 2 = Long).
+        /// Short mode maximum distance is limited to 1.3m but results in a better ambient immunity.
+        /// Long mode can range up to 4 m in the dark with a timing budget of200 ms.
+        /// </summary>
+        public Precision Precision
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP);
+                if (temp != 0x14 && temp != 0xA)
+                {
+                    return Precision.Unknown;
+                }
+
+                return temp == 0x14 ? Precision.Short : Precision.Long;
+            }
+            set
+            {
+                var timingBudget = TimingBudgetInMs;
+
+                switch (value)
+                {
+                    case Precision.Short:
+                        WriteRegister((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP, 0x14);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_A, 0x07);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_B, 0x05);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VALID_PHASE_HIGH, 0x38);
+                        WriteUInt16((ushort)Registers.SD_CONFIG__WOI_SD0, 0x0705);
+                        WriteUInt16((ushort)Registers.SD_CONFIG__INITIAL_PHASE_SD0, 0x0606);
+                        break;
+                    case Precision.Long:
+                        WriteRegister((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP, 0x0A);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_A, 0x0F);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_B, 0x0D);
+                        WriteRegister((ushort)Registers.RANGE_CONFIG__VALID_PHASE_HIGH, 0xB8);
+                        WriteUInt16((ushort)Registers.SD_CONFIG__WOI_SD0, 0x0F0D);
+                        WriteUInt16((ushort)Registers.SD_CONFIG__INITIAL_PHASE_SD0, 0x0E0E);
+                        break;
+                }
+
+                TimingBudgetInMs = timingBudget;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the intermeasurement period in ms.
+        /// Intermeasurement period must be >/= timing budget. This condition is not checked by the API,
+        /// the customer has the duty to check the condition. Default = 100 ms.
+        /// </summary>
+        public ushort InterMeasurementInMs
+        {
+            get
+            {
+                uint temp = ReadUInt32((ushort)Registers.VL53L1X_SYSTEM__INTERMEASUREMENT_PERIOD);
+                uint clockPll = ReadUInt32((ushort)Registers.VL53L1X_RESULT__OSC_CALIBRATE_VAL);
+
+                clockPll &= 0x3FF;
+
+                return (ushort)(temp / (clockPll * 1.065));
+            }
+            set
```

Comment:
nit, you need an extra CR between } and set.

Reply:
Done